File: Models/PhanQuyenTinhNang.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models.Database;

public partial class PhanQuyenTinhNang
{
    public int Id { get; set; }

    public int? NhomNguoiDung { get; set; }

    public string? DongHoTong { get; set; }

    public string? CauHinhDht { get; set; }

    public string? Dsdma { get; set; }

    public string? NrwcongTy { get; set; }

    public string? Nrwdma { get; set; }

    public string? DsngayDocSoBilling { get; set; }

    public string? NguoiDung { get; set; }

    public string? NhomNguoiDungTinhNang { get; set; }

    public string? NhatKySuDung { get; set; }

    public string? PhanQuyen { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;

    public virtual NhomNguoiDung? NhomNguoiDungNavigation { get; set; }
}

File: src/types/phan-quyen/phan-quyen-tinh-nang.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddPhanQuyenTinhNangRequest {
  nhomNguoiDung?: number;
  dongHoTong?: string;
  cauHinhDht?: string;
  dsdma?: string;
  nrwCongTy?: string;
  nrwDma?: string;
  dsngayDocSoBilling?: string;
  nguoiDung?: string;
  nhomNguoiDungTinhNang?: string;
  nhatKySuDung?: string;
  phanQuyen?: string;
  ngayTao: string;
  nguoiTao: number;
}

// Cập nhật dữ liệu
export interface UpdatePhanQuyenTinhNangRequest {
  nhomNguoiDung?: number;
  dongHoTong?: string;
  cauHinhDht?: string;
  dsdma?: string;
  nrwCongTy?: string;
  nrwDma?: string;
  dsngayDocSoBilling?: string;
  nguoiDung?: string;
  nhomNguoiDungTinhNang?: string;
  nhatKySuDung?: string;
  phanQuyen?: string;
  ngayCapNhat?: string;
  nguoiCapNhat?: string;

  // 👇 Cho phép key động
  [key: string]: string | number | undefined;
}

/* ========== RESPONSE (API --> CLIENT) ========== */
// Dữ liệu trả về
export interface PhanQuyenTinhNangResponse {
  id: number;
  nhomNguoiDung?: string;
  dongHoTong?: string;
  cauHinhDht?: string;
  dsdma?: string;
  nrwCongTy?: string;
  nrwDma?: string;
  dsngayDocSoBilling?: string;
  nguoiDung?: string;
  nhomNguoiDungTinhNang?: string;
  nhatKySuDung?: string;
  phanQuyen?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}
